src/library/scala/collection/immutable/RedBlackTree.scala

```diff
@@ -851,22 +851,22 @@ private[collection] object RedBlackTree {
 
   private[this] def _union[A, B](t1: Tree[A, B], t2: Tree[A, B])(implicit ordering: Ordering[A]): Tree[A, B] =
     if((t1 eq null) || (t1 eq t2)) t2
-    else if(t2 eq null) t1
+    else if(t1 eq null) t2
```

Comment:
should be 
`else if(t2 eq null) t1`

Reply:
This change was already merged in https://github.com/scala/scala/pull/9062, so it needs to be fixed in 2.12.x.

Reply:
https://github.com/scala/scala/pull/9075